rbc/omniscidb.py

```diff
@@ -757,151 +897,49 @@ def is_udtf(sig):
-                        sqlArgTypes = []
-                        for i, a in enumerate(sig[1]):
-                            _sizer = a.annotation().get('sizer')
-                            if _sizer is not None:
-                                # expect no more than one sizer argument
-                                assert sizer_index == -1
-                                sizer_index = i + 1
-                                sizer = _sizer
-                            atype = ext_arguments_map[
-                                a.tostring(use_annotation=False)]
-                            if 'output' in a.annotation():
-                                outputArgTypes.append(atype)
-                            else:
-                                if 'input' in a.annotation():
-                                    sqlArgTypes.append(atype)
-                                elif 'cursor' in a.annotation():
-                                    sqlArgTypes.append(
-                                        ext_arguments_map['Cursor'])
-                                inputArgTypes.append(atype)
-                        if sizer is None:
-                            sizer = 'kConstant'
-                        sizer_type = (thrift.TOutputBufferSizeType
-                                      ._NAMES_TO_VALUES[sizer])
-                        udtfs.append(thrift.TUserDefinedTableFunction(
-                            name + sig.mangling(),
-                            sizer_type, sizer_index,
-                            inputArgTypes, outputArgTypes, sqlArgTypes))
+                        udtfs_map[fid] = self._make_udtf_old(caller, orig_sig, sig)
                     else:
-                        rtype = ext_arguments_map[sig[0].tostring(
-                            use_annotation=False)]
-                        atypes = [ext_arguments_map[a.tostring(
-                            use_annotation=False)] for a in sig[1]]
-                        udfs.append(thrift.TUserDefinedFunction(
-                            name + sig.mangling(),
-                            atypes, rtype))
-                    signatures.append(sig)
+                        udfs_map[fid] = self._make_udf(caller, orig_sig, sig)
+                    signatures[fid] = sig
+
                 functions_and_signatures.append((caller.func, signatures))
-            llvm_module = compile_to_LLVM(functions_and_signatures,
-                                          target_info,
-                                          pipeline_class=OmnisciCompilerPipeline,
-                                          debug=self.debug)
 
-            self.catch_invalid_intrinsics(llvm_module, target_info)
+            llvm_module, succesful_fids = compile_to_LLVM(
+                functions_and_signatures,
+                target_info,
+                pipeline_class=OmnisciCompilerPipeline,
+                debug=self.debug)
+
+            # self.catch_invalid_intrinsics(llvm_module, target_info)
```

Comment:
Is the idea to prevent GPU code from calling libm/stdlib functions? If yes, then, you can remove `catch_invalid_intrinsics` call and implementation.

Reply:
Removed.